cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/generator/CobiGenWrapper.java

```diff
@@ -544,10 +544,12 @@ public boolean isMergableFile(String path, Collection<IncrementTo> consideredInc
      *         <code>false</code> otherwise
      */
     private boolean isMergableFile(Set<TemplateTo> templates) {
-        for (TemplateTo template : templates) {
-            if (template.getMergeStrategy() != null
-                && !template.getMergeStrategy().equals(ConfigurationConstants.MERGE_STRATEGY_OVERRIDE)) {
-                return true;
+        if (templates != null) {
```

Comment:
I am not totally sure, whether this solves the issue correctly.
I would guess you got a null pointer exception while referring to an increment of another trigger, which @jdiazgon initially implemented. For any reason, the list of templates is empty, which for me seems to be more of an invalid state of the application. Maybe the templates of the referenced increment has been tried to accessed and could not determined. Just checking not null here would mean, that you will not determine whether a path in the preview is going to be merged or not. Most probably by your fix, every path resulting from a externally referenced increment is shown as non-mergable, although it might be.

Reply:
> I am not totally sure, whether this solves the issue correctly.

You are right. Actually this was my idea, to simply add the `if` condition as a workaround. The thing is that Mandy and me tried to debug this but we were stuck, so I decided that maybe for this kind of small issues, we should not spent too much time.

The thing is that before adding the `if`, a window with a NPE was popping up when trying to generate code. But this NPE was not harmful at all because anyway it generated all the templates.

Reply:
You are right. It will generate, but the merge / generate preview will not show the truth.
So for me its a more severe problem as we are playing with the trust of the people to use cobigen.

Reply:
True. When I have time, I will take a look into this.